py4DSTEM/visualize/show.py

```diff
@@ -247,9 +252,10 @@ def show(ar,figsize=(8,8),cmap='gray',scaling='none',clipvals='minmax',
         _ar = ar.copy()
         _mask = np.ones_like(_ar.data,dtype=bool)
     elif scaling == 'log':
-        _mask = ar>0
+        _mask = ar.data>0
         _ar = np.zeros_like(ar.data,dtype=float)
-        _ar[_mask] = np.log(ar[_mask])
+        _ar[_mask] = np.log(ar.data[_mask])
+        _ar[~_mask] = -np.inf
```

Comment:
why -inf and not nan?

Reply:
That's... a good question. I think that NaN would work just as well and is probably the more sensible choice. I will change these.